File: distribution/python/dqops/client/models/import_schema_queue_job_parameters.py
from typing import Any, Dict, List, Type, TypeVar, Union

from attrs import define as _attrs_define
from attrs import field as _attrs_field

from ..types import UNSET, Unset

T = TypeVar("T", bound="ImportSchemaQueueJobParameters")


@_attrs_define
class ImportSchemaQueueJobParameters:
    """
    Attributes:
        connection_name (Union[Unset, str]): Connection name where the tables are imported.
        schema_name (Union[Unset, str]): Source schema name from which the tables are imported.
        table_name_contains (Union[Unset, str]): Optional filter for the names of tables to import, it is a text
            (substring) that must be present inside table names. This filter is case sensitive.
    """

    connection_name: Union[Unset, str] = UNSET
    schema_name: Union[Unset, str] = UNSET
    table_name_contains: Union[Unset, str] = UNSET
    additional_properties: Dict[str, Any] = _attrs_field(init=False, factory=dict)

    def to_dict(self) -> Dict[str, Any]:
        connection_name = self.connection_name
        schema_name = self.schema_name
        table_name_contains = self.table_name_contains

        field_dict: Dict[str, Any] = {}
        field_dict.update(self.additional_properties)
        field_dict.update({})
        if connection_name is not UNSET:
            field_dict["connectionName"] = connection_name
        if schema_name is not UNSET:
            field_dict["schemaName"] = schema_name
        if table_name_contains is not UNSET:
            field_dict["tableNameContains"] = table_name_contains

        return field_dict

    @classmethod
    def from_dict(cls: Type[T], src_dict: Dict[str, Any]) -> T:
        d = src_dict.copy()
        connection_name = d.pop("connectionName", UNSET)

        schema_name = d.pop("schemaName", UNSET)

        table_name_contains = d.pop("tableNameContains", UNSET)

        import_schema_queue_job_parameters = cls(
            connection_name=connection_name,
            schema_name=schema_name,
            table_name_contains=table_name_contains,
        )

        import_schema_queue_job_parameters.additional_properties = d
        return import_schema_queue_job_parameters

    @property
    def additional_keys(self) -> List[str]:
        return list(self.additional_properties.keys())

    def __getitem__(self, key: str) -> Any:
        return self.additional_properties[key]

    def __setitem__(self, key: str, value: Any) -> None:
        self.additional_properties[key] = value

    def __delitem__(self, key: str) -> None:
        del self.additional_properties[key]

    def __contains__(self, key: str) -> bool:
        return key in self.additional_properties

File: dqops/src/main/java/com/dqops/core/jobqueue/jobs/schema/ImportSchemaQueueJobParameters.java
/*
 * Copyright © 2021-Present DQOps, Documati sp. z o.o. (support@dqops.com)
 *
 * This file is licensed under the Business Source License 1.1,
 * which can be found in the root directory of this repository.
 *
 * Change Date: This file will be licensed under the Apache License, Version 2.0,
 * four (4) years from its last modification date.
 */
package com.dqops.core.jobqueue.jobs.schema;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;
import lombok.EqualsAndHashCode;

/**
 * Parameters for the {@link ImportSchemaQueueJob} job that imports tables from a database.
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
@EqualsAndHashCode(callSuper = false)
public class ImportSchemaQueueJobParameters {
    @JsonPropertyDescription("Connection name where the tables are imported.")
    private String connectionName;

    @JsonPropertyDescription("Source schema name from which the tables are imported.")
    private String schemaName;

    @JsonPropertyDescription("Optional filter for the names of tables to import, it is a text (substring) that must be present inside table names. This filter is case sensitive.")
    private String tableNameContains;

    public ImportSchemaQueueJobParameters() {
    }

    /**
     * Creates a schema import job parameters.
     * @param connectionName Connection name to import.
     * @param schemaName Schema name in the source database to import.
     * @param tableNameContains Optional table name search filter.
     */
    public ImportSchemaQueueJobParameters(String connectionName, String schemaName, String tableNameContains) {
        this.connectionName = connectionName;
        this.schemaName = schemaName;
        this.tableNameContains = tableNameContains;
    }

    /**
     * Returns the connection name to be imported.
     * @return Connection name.
     */
    public String getConnectionName() {
        return connectionName;
    }

    /**
     * Sets the connection name to be imported.
     * @param connectionName Connection name.
     */
    public void setConnectionName(String connectionName) {
        this.connectionName = connectionName;
    }

    /**
     * Returns the source schema name.
     * @return Source schema name.
     */
    public String getSchemaName() {
        return schemaName;
    }

    /**
     * Sets the source schema name.
     * @param schemaName Source schema name.
     */
    public void setSchemaName(String schemaName) {
        this.schemaName = schemaName;
    }

    /**
     * Sets an optional table filter that should be a text inside the table name, without any wildcard characters.
     * @return Table name substring.
     */
    public String getTableNameContains() {
        return tableNameContains;
    }

    /**
     * Sets the table name search filter.
     * @param tableNameContains Table name search filter.
     */
    public void setTableNameContains(String tableNameContains) {
        this.tableNameContains = tableNameContains;
    }
}
